packages/@aws-cdk/aws-codedeploy/lib/lambda/deployment-config.ts

```diff
@@ -0,0 +1,14 @@
+/**
+ * Deployment Config for Lambda CodeDeploy Applications.
+ */
+export enum LambdaDeploymentConfig {
+  AllAtOnce = 'AllAtOnce',
+  Canary10Percent30Minutes = 'Canary10Percent30Minutes',
+  Canary10Percent5Minutes = 'Canary10Percent5Minutes',
+  Canary10Percent10Minutes = 'Canary10Percent10Minutes',
+  Canary10Percent15Minutes = 'Canary10Percent15Minutes',
+  Linear10PercentEvery10Minutes = 'Linear10PercentEvery10Minutes',
+  Linear10PercentEvery1Minute = 'Linear10PercentEvery1Minute',
+  Linear10PercentEvery2Minutes = 'Linear10PercentEvery2Minutes',
+  Linear10PercentEvery3Minutes = 'Linear10PercentEvery3Minutes'
+}
```

Comment:
This is not enough. You can also have custom Lambda Deployment Configurations.

Please model them in a similar way to how the Server Deployment Configurations are done.

Reply:
Can you provide the documentation? I could not find anything describing this in full detail, everything pointed to SAM.

Reply:
Sorry, I should have been more precise in my comment.

Yes, this is currently a gap in CodeDeploy's CFN support. However, at some point, it's going to be supported (you can create custom Deployment Configurations in the console), and we should make adding that support backwards-compatible. So, we need to future proof this solution, and make the default Configurations available through static fields, similarly like it's done currently for Server Deployment Configurations. An `enum` won't cut it.